### DISCUSSION TITLE
Issues Tools uiProvider not displayed in custom iTwin viewer

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#1842] [Q&A] [answered]
In **_Design Review_** I see a different set of uiProviders than I do in my custom iTwin viewer.  See the two images below.

In particular I'd like to provide the **_Issue Tools_** button.  Is this something that can be configured in my viewer or is it a feature specific to _**Design Review?**_

Thanks,

John M.

**Design Review**
![image](https://user-images.githubusercontent.com/30668739/125309862-b405a380-e2f7-11eb-81e6-c05e7dca0891.png)

**My custom iTwin viewer**
![image](https://user-images.githubusercontent.com/30668739/125310404-28d8dd80-e2f8-11eb-8122-b3bd58d3c955.png)

**Comment by aruniverse** [comment, marked as answer]
Hi John, 

At the moment the Issues Tools you're seeing in Design Review is not available to consume in 3rd party Viewer apps. We'll look into open sourcing that for external users.

I'd recommend taking a look at the [Issues Sample](https://www.itwinjs.org/sample-showcase/?group=iTwin+Platform+APIs&sample=issue-sample&imodel=Metrostation+Sample) to build your own

**Reply by JohnMajerle** [reply]
Hello Arun,

For what its worth, that sample hangs when I go to your link:

![image](https://user-images.githubusercontent.com/30668739/125352612-1b871780-e327-11eb-9af5-096d0f20d254.png)

-- John M.

**Reply by aruniverse** [reply]
Sorry about that John, I just reproduced the same issue. 

@StefanRetief @dustinlebsock can you guys help look into this?

**Reply by aruniverse** [reply]
@JohnMajerle could you change the `baseUrl` in `IssuesClient.ts` to drop the `/v1` then click on the the green run button in the top right corner of the editor to apply your changes. This should correctly fetch the issues.

![image](https://user-images.githubusercontent.com/11051042/125356907-de298680-e334-11eb-89b5-f2a2a3154b5c.png)
![image](https://user-images.githubusercontent.com/11051042/125356986-fa2d2800-e334-11eb-8a82-447300f74f9c.png)
![image](https://user-images.githubusercontent.com/11051042/125357102-2b0d5d00-e335-11eb-9b90-91dda186b54b.png)

I'll coordinate with the team responsible to make sure these changes are deployed asap.

**Reply by JohnMajerle** [reply]
@aruniverse 

Hello Arun,

I attempted to pull in the **_Issues Sample_** code into my Electron desktop viewer that I created using **_desktop-starter._**   Unfortunately the provided file **_IssuesClient.ts_** references an npm package **_@itwinjs-sandbox_** that doesn't appear to exist:

`import { AuthorizationClient } from "@itwinjs-sandbox";`

To get around this I tried to modifying getAccessToken to use my existing client:

```
    private static async getAccessToken() {
      try {
        return (IModelApp.authorizationClient as FrontendAuthorizationClient).getAccessToken();
      } catch (e) {
        return undefined;
    }
```

...but that doesn't work due to CORS errors.  

Can you suggest what I need to do to correct this?

Thanks,

John M.

**Reply by aruniverse** [reply]
Hey john,

This is the same issue as mentioned in this discussion post: https://github.com/imodeljs/imodeljs/discussions/1699
Basically to use our apis available at developer.bentley.com you need to use an auth token issued by `ims.bentley.com`; where as by default the token you obtain from `IModelApp.authorizationClient.getAccessToken()` is issued by `imsoidc.bentley.com`.

We're trying to consolidate the two and make it easier so you don't have to manage 2 different oidc-clients yourself, and should have a public solution for this in a few weeks. 

I did see your question on that discussion post about how to handle this on a desktop based application. I've been playing around with it this week, and think i have a solution, but I'm no desktop expert. I'll be chatting with someone more familiar with that early next week and will get back to you.

**Reply by aruniverse** [reply]
Hey john, 

Spoke with a few of our desktop experts, and unfortunately we don't have a public solution for you yet. We expect to push the solution to prod next week, and hope you can wait til then. Once that change is available, ill provide a sample that you can use.
